openslides/core/static/js/core/base.js

```diff
@@ -553,27 +553,57 @@ angular.module('OpenSlidesApp.core', [
 ])
 
 // Template hooks
-// 2 possible uses:
-// - { Id: 'myHookId', template: '<button>click me</button>' }
-// - { Id: 'myHookId', templateUrl: '/static/templates/plugin_name/my-hook.html' }
-// It is possible to provide a scope, that is merged into the scope of the templateHook.
-// This overrides functions/values of the parent scope, but there may are conflicts
-// with other plugins defining the same function/value. E.g.:
+// Possible uses:
+// 1. { id: 'myHookId', template: '<button>click me</button>' }
+// 2. { id: 'myHookId', templateUrl: '/static/templates/plugin_name/my-hook.html' }
+// 3. { id: 'myHookId' }
+//
+// Deprecated: Give the id with 'Id'. Please use 'id'.
+//
+// Option 3 is for just changing the scope (see below), but not the original content. This
+// is usefull to alter a JS behavior, e.g. on a ng-click. In this case, override is false
+// for this template hook.
+//
+// It is possible to provide a scope, that is merged into the surrounding scope.
+// You can override functions or values of the surrounding scope by providing them:
 // { Id: 'hookId', template: '<button ng-click="customFn()">click me</button>',
 //   scope: {
-//     customFn: function () { /*Do something */ },
+//     customOrOverwritten: function () { /*Do something */ },
 //   },
 // }
+// Or you provide a function that returns an object of functions/values to overwrite to
+// get access to the scope merged in:
+// { Id: 'hookId', template: '<button ng-click="customFn()">click me</button>',
+//   scope: function (scope) {
+//     return {
+//       customOrOverwritten: function () {
+//         scope.value = /* change it */;
+//       },
+//     };
+//   },
+// }
+//
+// As a default, template hooks in flavour of option 1 and 2 override the content that was
+// originally there. Provide 'override: false', to prevent overriding the original content.
 .factory('templateHooks', [
     function () {
         var hooks = {};
         return {
             hooks: hooks,
             registerHook: function (hook) {
-                if (hooks[hook.Id] === undefined) {
-                    hooks[hook.Id] = [];
+                // Deprecated: Set the new style 'id', if 'Id' is given.
+                if (hook.id === undefined) {
```

Comment:
Please use `void 0` instead of `undefined`. (I see that there was `undefined` before ... ;-))